Mypy failing on util/_download.py file

### What happened?

Mypy is currently failing CI checks on devel within the `util/_download.py` file.
```
openghg/util/_download.py:86: error: Cannot assign to a method  [method-assign]
openghg/util/_download.py:88: error: Argument 1 to "wrap_file" has incompatible type "HTTPResponse | Any"; expected "BinaryIO"  [arg-type]
```
I have not currently been able to reproduce this locally and the mypy tests all pass.

### What did you expect to happen?

Mypy tests to all pass.

### Minimal Complete Verifiable Example

```Python
Running the CI on devel which includes mypy run.
```


### Relevant log output

_No response_

### Anything else we need to know?

_No response_

### Environment

<details>



</details>

Creating a clean virtual environment I can now replicate this issue: [environment_details.txt](https://github.com/user-attachments/files/16929354/environment_details.txt). I'm not currently sure why this new environment now causes the mypy fails as there are no obvious upgrades to packages which would cause this.

In relation to the two errors for:
```
openghg/util/_download.py:86: error: Cannot assign to a method  [method-assign]
```

The relevant line of code is
```
r.raw.read = functools.partial(r.raw.read, decode_content=True)
```

The addition of this line could be related to the issue: [no way to read uncompressed content as file-like object](https://github.com/psf/requests/issues/465) related to the requests module as this contains an example of the line of code above as a suggestion. It may be that this is not needed anymore?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -20,6 +20,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Bug where an input filepath list to standardise_surface was only storing the last file hash. This allowed for some files to bypass the check for the same files depending on where they were in the original filepath list. [PR #1100](https://github.com/openghg/openghg/pull/1100)
 - Bug where filepath needed to be a Path object when storing the file hash values. [PR #1108](https://github.com/openghg/openghg/pull/1108)
 - Catch an `AttributeError` when trying synchronise attributes and metadata and a user passes a `bool` - [PR #1029](https://github.com/openghg/openghg/pull/1029)
+- Mypy issue fixed for `util.download_data()` function based on updates described [requests Issue 465](https://github.com/psf/requests/issues/465) and included in [urllib3 PR 159](https://github.com/urllib3/urllib3/pull/159/files). This allowed the `decode_content` flag to be set directly rather than needing to patch the method. [PR #1118](https://github.com/openghg/openghg/pull/1118)
+
 
 ## [0.9.0] - 2024-08-14
 

--- a/openghg/util/_download.py
+++ b/openghg/util/_download.py
@@ -34,7 +34,6 @@ def download_data(
     Returns:
         bytes / None: Bytes if no filepath given
     """
-    import functools
     import io
     import shutil
     from urllib.parse import urlparse

--- a/openghg/util/_download.py
+++ b/openghg/util/_download.py
@@ -83,9 +83,11 @@ def download_data(
     file_size = int(r.headers.get("Content-Length", 0))
 
     desc = f"Downloading {filename}"
-    r.raw.read = functools.partial(r.raw.read, decode_content=True)
+    r.raw.decode_content = True
 
-    with wrap_file(r.raw, total=file_size, description=desc) as r_raw:
+    # mypy error ignored
+    # rich and requests libraries not quite aligning but urllib3.response.HTTPResponse should be very similiar to BinaryIO object expected.
+    with wrap_file(r.raw, total=file_size, description=desc) as r_raw:  # type:ignore
         with io.BytesIO() as buf:
             shutil.copyfileobj(r_raw, buf)
 

